projects/go-lib/src/lib/components/go-timepicker/go-time.component.ts

```diff
@@ -0,0 +1,230 @@
+
+import { GoTime } from './go-time';
+@Component({
+  selector: 'go-time',
+  styleUrls: ['./go-time.component.scss'],
+  templateUrl: './go-time.component.html',
+  animations: [fadeAnimation],
+})
+export class GoTimeComponent implements OnInit {
+  @Input() appendToContent: boolean;
+  @Input() goTime: GoTime;
+  @Input() displayAbove: boolean;
+  @Input() displayFromRight: boolean;
+
+  @Output() timePicked: any = new EventEmitter<any>();
+  inputHourError: boolean = false;
+  inputMinuteError: boolean = false;
+  format: boolean = true;
+  opened: boolean = false;
+  clickInside: boolean = false;
+  clickActionButton: boolean = false;
+  timeFormat: any = { hours: '', minutes: '', ampm: '' };
+  subscription: any;
+  hour: any;
+  minute: any;
+
+  constructor() {}
+
+  @HostListener('click')
+  ClickInside(): void {
+    this.clickInside = this.clickActionButton ? false : true;
+    this.opened = true;
+  }
+
+  @HostListener('document: click')
+  onDocumentClick(): void {
+    if (this.opened && !this.clickInside) {
+      this.closeTimePicker();
+      this.resetTimeInput();
+    }
+    this.clickInside = false;
+    this.clickActionButton = false;
+  }
+  @HostListener('window:keydown', ['$event'])
+  keyEvent(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'Esc': // IE/Edge specific value
+      case 'Escape':
+        event.preventDefault();
+        this.closeTimePicker();
+        this.resetTimeInput();
+        break;
+      default:
+        return;
+    }
+  }
+  public closeTimePicker(): void {
+    this.goTime.closeTime();
+  }
+  ngOnInit(): void {
+    this.subscription = this.goTime.timeWindowOpen.subscribe(
+      (value: boolean) => {
+        if (value) {
+          if (this.goTime.selectedTime) {
+            this.changeTimeFormat(this.goTime.selectedTime);
+          } else {
+            this.timeFormat = this.formatAMPM(new Date());
+            this.hour = this.timeFormat.hours;
+            this.minute = this.timeFormat.minutes;
+            this.format = this.timeFormat.ampm === 'am' ? true : false;
+          }
+          this.autoFocousInput();
+        }
+        this.opened = value;
+      }
+    );
+  }
+
+  autoFocousInput(): void {
+    setTimeout(() => {
+      document.getElementById('hour-input').focus();
+    });
+  }
+
+  formatAMPM(date: any): void {
+    let hours: any = date.getHours();
+    let minutes: any = date.getMinutes();
+    const ampm: any = hours >= 12 ? 'pm' : 'am';
+    hours = hours % 12;
+    hours = hours ? hours : 12; // the hour '0' should be '12'
+    hours = hours > 9 ? hours : '0' + hours;
+    minutes = minutes > 9 ? minutes : '0' + minutes;
+    const formatTime: any = { hours, minutes, ampm };
+    return formatTime;
+  }
+  changeTimeFormat(timeString: any): void {
+    const H: any = +timeString.substr(0, 2);
+    const h: any = H % 12 || 12;
+    const ampm: any = H < 12 || H === 24 ? 'AM' : 'PM';
+    const minute: any = timeString.substr(3, 2);
```

Comment:
Can we add a type here?

Reply:
I have an update on that code. Please check once you are looking like this or not.